server/src/testFixtures/java/io/crate/testing/auth/TestJWTAuthenticationMethod.java

```diff
@@ -0,0 +1,144 @@
+import io.crate.role.Roles;
+
+/**
+ * Test implementation, allowing to supply own keys.
+ * <br />
+ *
+ * Public key and token, signed by its counterpart private key
+ * are generated via https://jwt.io/#debugger-io.
+ * Token contains dummy key id (kid) which is assigned to a public key.
+ * exp is NumericDate which is epoch seconds, see https://www.rfc-editor.org/rfc/rfc7519#section-4.1.4
+ * <br />
+ *
+ * Decoded token:
+ * Header:
+ * {
+ *   "alg": "RS256",
+ *   "typ": "JWT",
+ *   "kid": "1"
+ * }
+ * Payload:
+ * {
+ *   "iss": "https://console.cratedb-dev.cloud/api/v2/meta/jwk/",
+ *   "username": "cloud_user",
+ *   "exp": 1893456000
+ * }
+
+ */
+public class TestJWTAuthenticationMethod extends JWTAuthenticationMethod {
+
+    // A dummy value which would normally pass in all happy path scenario.
+    public static final LocalDateTime EXPIRES_AT = LocalDate.of(2030, Month.JANUARY, 1).atStartOfDay();
+
+    public static final String JWT_TOKEN = """
+        eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IjEifQ.eyJpc3MiOiJod\
+        HRwczovL2NvbnNvbGUuY3JhdGVkYi1kZXYuY2xvdWQvYXBpL3YyL21ldGEvandrL\
+        yIsInVzZXJuYW1lIjoiY2xvdWRfdXNlciIsImV4cCI6MTg5MzQ1NjAwMH0.G2Z2c\
+        lzQsx_HFOlBmqMCfuJKQYljxOqlZGiGyWgkDbSJLNKyL_zgI8AtuGj9BT6cM_bng\
+        dA5v6z8aKIdCu0KDRJrfAa-fRtALEDhlcE8hJoNy8dFXdj41YSdJW0Kb-1Lwos4J\
+        2_LE70-0f6-J7Z0ssRbirwgwgH-xq_sdW9t97RJhgAjoGn9otOwVzfBgS3rpx_mf\
+        Qyy-oye25noQTPP7A4gGO8X_0rCP0qGzkmYjhYluzJRavNeylD7Cqjm-pbqE3tfB\
+        -RSViBWVqBd-ymMyMVyV-Xa7uxKZaVXAStZ2erNpla_TTM9SWCQ3inPeXMuc6PVj\
+        XME2b_KIY6JVTBbuA\
+        """;
+
+    public static Role JWT_USER = userOf(
+        "John",
+        Set.of(),
+        new HashSet<>(),
+        getSecureHash("johns-pwd"),
+        new JwtProperties("https://console.cratedb-dev.cloud/api/v2/meta/jwk/", "cloud_user")
+    );
+
+    private static final String KID = "1";
+
+    private final JwkProvider mockJwkProvider = mock(JwkProvider.class);
+
+    private final Jwk mockJwk = mock(Jwk.class);
+
+    private final Clock clock;
+    private final String algorithm;
+
+    public TestJWTAuthenticationMethod(Roles roles, @Nullable Clock customClock, @Nullable String algorithm) {
+        super(roles);
+        this.clock = customClock != null ? customClock : Clock.systemUTC();
+        this.algorithm = algorithm;
+        initMocks();
+    }
+
+    private void initMocks() {
+        try {
+            EncodedKeySpec publicKeySpec = new X509EncodedKeySpec(Base64.getDecoder().decode(PUBLIC_KEY_256));
+            KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+            PublicKey publicKey = keyFactory.generatePublic(publicKeySpec);
+
+            when(mockJwkProvider.get(KID)).thenReturn(mockJwk);
+            when(mockJwk.getPublicKey()).thenReturn(publicKey);
+            when(mockJwk.getAlgorithm()).thenReturn(algorithm);
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
```

Comment:
Why do we need the mocking and a custom Test authentication method?
Makes it way harder to see what's actually being tested.

Are the tests that go through the http handlers not sufficient?
Could we instead parameterize the `JwkProvider` for `JWTAuthenticationMethod` ?

Reply:
Non mocked `JwkProvider` in `JWTAuthenticationMethod` would try to access an endpoint (so would need to setup endpoint in all relevant unit tests)

> Could we instead parametrize the JwkProvider for JWTAuthenticationMethod?

I will add an interface for jwk/clock providers and get rid of `TestJWTAuthenticationMethod`

Reply:
We need mocking to avoid http call (at least in tests for `JWTAuthenticationMethod` -> new cases in `UserAuthenticationMethodTest`
Got rid of `TestJwtAuthenticaitonMethod` - instead made `JWTAuthenticationMethod` configurable in
https://github.com/crate/crate/pull/15551/commits/8d22388c7d2f090042deb31318d81a7a901499db